CONTRIBUTIONS.md

```diff
@@ -54,33 +54,40 @@ A lot of people are excited about building a JSON API in Servant and everyone se
 
 ## UI languages/frameworks
 
-| Library/framewor | Who's working on it |
-| --- | --- |
-| Elm | No one yet |
-| Purescript | No one yet |
-| Reflex FRP | No one yet |
-
+| Library/framework | Who's working on it                     |
+| ---               | ---                                     |
+| Elm               | No one yet                              |
+| Purescript        | No one yet                              |
+| Reflex FRP        | [meditans](https://github.com/meditans) |
 
 # Higher order design goals
 
+## General principles
+* Making nonsensical states non-representible in the domain model. eg. product can have only two type -- physical and digital. Status can have only few values. Basically a lot of ADT usage.
+* Lifting more invariants to the type-level
+* Unit tests
+* Controller tests
+* Integration tests
```

Comment:
From a practical standpoint, right now I feel that integration tests are the **most important** aspect of UI testing. For our current project, what kind of stuff can be tested in the unit & controller tests at the **UI layer?**